pkg/manager/impl/executions/phase_filters.go

```diff
@@ -0,0 +1,119 @@
+package executions
+
+import (
+	"github.com/flyteorg/flyteadmin/pkg/common"
+	"github.com/flyteorg/flyteadmin/pkg/errors"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"google.golang.org/grpc/codes"
+)
+
+const phaseField = "phase"
+
+var TerminalWorkflowExecutionPhases = []string{
+	core.WorkflowExecution_ABORTED.String(),
+	core.WorkflowExecution_FAILED.String(),
+	core.WorkflowExecution_TIMED_OUT.String(),
+	core.WorkflowExecution_SUCCEEDED.String(),
+}
+
+var SchedulingWorkflowPhases = []string{
+	core.WorkflowExecution_UNDEFINED.String(),
+	core.WorkflowExecution_QUEUED.String(),
+}
+
+var TerminalNodeExecutionPhases = []string{
+	core.NodeExecution_SUCCEEDED.String(),
+	core.NodeExecution_FAILED.String(),
+	core.NodeExecution_TIMED_OUT.String(),
+	core.NodeExecution_ABORTED.String(),
+	core.NodeExecution_SKIPPED.String(),
+	core.NodeExecution_RECOVERED.String(),
+}
+
+var TerminalTaskExecutionPhases = []string{
+	core.TaskExecution_SUCCEEDED.String(),
+	core.TaskExecution_FAILED.String(),
+	core.TaskExecution_ABORTED.String(),
+}
+
+func newNotTerminalFilter(entity common.Entity) (common.InlineFilter, error) {
+	switch entity {
+	case common.Execution:
+		return common.NewRepeatedValueFilter(common.Execution, common.ValueNotIn, "phase", TerminalWorkflowExecutionPhases)
+	case common.NodeExecution:
+		return common.NewRepeatedValueFilter(common.Execution, common.ValueNotIn, "phase", TerminalNodeExecutionPhases)
+	case common.TaskExecution:
+		return common.NewRepeatedValueFilter(common.Execution, common.ValueNotIn, "phase", TerminalTaskExecutionPhases)
+	default:
+		return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Unrecognized execution entity [%+v] for non-terminal filters", entity)
+	}
+}
+
+func newSchedulingFilter() (common.InlineFilter, error) {
+	return common.NewRepeatedValueFilter(common.Execution, common.ValueIn, "phase", SchedulingWorkflowPhases)
+}
+
+func GetUpdateExecutionFilters(eventPhase core.WorkflowExecution_Phase) (filters []common.InlineFilter, err error) {
+	// With the exeception of queued events, it's never acceptable to move into the same phase.
```

Comment:
Hmmm I'm not sure this is accurate. If propeller picks up a WF, sends running event then fails to write the CRD back to ApiServer. It'll attempt to do the same operation again... why is this a bad thing not just an idempotent operation?

Reply:
oh interesting, the current implementation doesn't let us update to an existing phase. i thought propeller swallowed the already exists in that case e.g. https://github.com/flyteorg/flyteadmin/blob/master/pkg/manager/impl/execution_manager.go#L1183
andhttps://github.com/flyteorg/flytepropeller/blob/c016dabbfef6037bead59590b42326dabe89f957/pkg/controller/workflow/executor.go#L261

perhaps propeller should also be modified to swallow the NotFound when RowsAffected == 0 gets triggered as part of meeting the new phase conditions. but perhaps that is getting too finnicky